### PR TITLE
[INLONG-8681][Security] Upgrade io.netty:netty-handler to version 4.1.94.Final

### PR DESCRIPTION
**Pull request by SCUTking**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- Title Example: [INLONG-XYZ][Component] Title of the pull request

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes https://github.com/apache/inlong/issues/8681

### Motivation

The SniHandler can allocate up to 16MB of heap for each channel during the TLS handshake. When the handler orthe channel does not have an idle timeout, it can be used to make a TCP server using the SniHandler to allocate
16MB of heap.

### Modifications

Upgrade io.netty:netty-handler to version 4.1.94 and edit some LICENSE

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -647,7 +647,7 @@
             <dependency>
                 <groupId>io.netty</groupId>
                 <artifactId>netty-handler</artifactId>
-                <version>${netty.version}</version>
+                <version>4.1.94.Final</version>
```
Comment (healchow) [inline pom.xml:650]: Execuse me, byt why not change the `${netty.version}` to `4.1.94.Final` directly?

Comment (healchow) [reply pom.xml:650]: Maybe you can add a new variable like `${netty.handler.version}` instead of directly using it in the dependency version.

Comment (SCUTking) [reply pom.xml:650]: Got it, I'll go modify it.

Comment (dockerzhang) [reply pom.xml:650]: it's better to update `netty.version` to `4.1.94.Final ` other than only `handler`.

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**